YARCH/Sources/Modules/CatalogDetails/CatalogDetailsPresenter.swift

```diff
@@ -0,0 +1,61 @@
+// Отвечает за отображение данных модуля CatalogDetails
+
+import UIKit
+
+protocol CatalogDetailsPresentationLogic {
+	func presentFetchedDetails(response: CatalogDetails.FetchDetails.Response)
+    func presentOpenExternalLink(response: CatalogDetails.OpenExternalLink.Response)
+}
+
+class CatalogDetailsPresenter: CatalogDetailsPresentationLogic {
+
+	weak var viewController: CatalogDetailsDisplayLogic?
+
+    let numberFormatterWorker: NumberFormatterWorker
+
+    init(numberFormatterWorker: NumberFormatterWorker = NumberFormatterWorker()) {
+        self.numberFormatterWorker = numberFormatterWorker
+    }
+
+	// MARK: Present Fetched Details
+
+	func presentFetchedDetails(response: CatalogDetails.FetchDetails.Response) {
+		var viewModel: CatalogDetails.FetchDetails.ViewModel
+
+        switch response.result {
+        case let .success(model, imageData):
+            let coinSnapshotViewModel = makeCoinSnapshotViewModel(model, imageData: imageData)
+            viewModel = CatalogDetails.FetchDetails.ViewModel(coinId: model.uid, error: nil, snapshotViewModel: coinSnapshotViewModel, infoRepresentable: coinSnapshotViewModel.properties)
+        case let .failure(id, error):
+            viewModel = CatalogDetails.FetchDetails.ViewModel(coinId: id, error: error, snapshotViewModel: nil, infoRepresentable: nil)
+        }
+        viewController?.displayFetchedDetails(viewModel: viewModel)
+	}
+
+    // MARK: Present Open External Link
+
+    func presentOpenExternalLink(response: CatalogDetails.OpenExternalLink.Response) {
+        switch response.result {
+        case let .success(url):
+            viewController?.displayOpenExtenalLink(viewModel: CatalogDetails.OpenExternalLink.ViewModel(url: url, error: nil))
+        case let .failure(error):
+            viewController?.displayOpenExtenalLink(viewModel: CatalogDetails.OpenExternalLink.ViewModel(url: nil, error: error))
+        }
+    }
+
+    func makeCoinSnapshotViewModel(_ model: CoinSnapshotFullModel, imageData: Data?) -> CoinSnapshotFullViewModel {
+        let title = model.title
+        let image = UIImage(data: imageData ?? Data())
+        let website = CoinSnapshotPropertyViewModel(type: .website, value: "Click here to open URL")
```

Comment:
Maybe it's better to use `model.website` like `model.twitter`?

Reply:
Made it a bit clearly – we need to show annotation message at first and make another call in order to resolve an actual URL